File: app/src/main/java/xyz/ummo/user/ui/fragments/profile/ProfileViewModel.java
package xyz.ummo.user.ui.fragments.profile;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import timber.log.Timber;
import xyz.ummo.user.data.entity.ProfileEntity;
import xyz.ummo.user.data.repo.AppRepository;

public class ProfileViewModel extends AndroidViewModel {

    private AppRepository appRepository;
    private LiveData<ProfileEntity> profileEntityLiveData;

    public ProfileViewModel(@NonNull Application application) {
        super(application);

        appRepository = new AppRepository(application);
        profileEntityLiveData = appRepository.getProfileEntityLiveData();
    }

    public void insertProfile(ProfileEntity profileEntity){
        appRepository.insertProfile(profileEntity);
        Timber.e("insertProfile: PROFILE->%s", profileEntity.getProfileName());
    }

    public void updateProfile(ProfileEntity profileEntity){
        appRepository.updateProfile(profileEntity);
    }

    public void deleteAllProfile(){
        appRepository.deleteProfile();
    }

    public LiveData<ProfileEntity> getProfileEntityLiveData(){
        return appRepository.getProfileEntityLiveData();
    }
}

File: app/src/main/java/xyz/ummo/user/ui/fragments/profile/ProfileViewModel.kt
package xyz.ummo.user.ui.fragments.profile

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.work.OneTimeWorkRequest
import androidx.work.WorkManager
import xyz.ummo.user.data.entity.ProfileEntity
import xyz.ummo.user.data.repo.AppRepository
import xyz.ummo.user.workers.SocketConnectWorker

class ProfileViewModel(application: Application) : AndroidViewModel(application) {
    private val appRepository: AppRepository = AppRepository(application)

    /** Creating an instance of WorkManager for SocketConnectWorker **/
    private val socketConnectWorkManager = WorkManager.getInstance(application)

    val profileEntityLiveData: LiveData<ProfileEntity> = appRepository.profileEntityLiveData
    val profileEntityListData: List<ProfileEntity> = appRepository.profileEntityListData

    internal fun socketConnect() {
        socketConnectWorkManager.enqueue(OneTimeWorkRequest.from(SocketConnectWorker::class.java))
    }

    fun insertProfile(profileEntity: ProfileEntity?) {
        appRepository.insertProfile(profileEntity)
        //        Timber.e("insertProfile: PROFILE->%s", profileEntity.profileName);
    }

    fun updateProfile(profileEntity: ProfileEntity?) {
        appRepository.updateProfile(profileEntity)
    }

    fun deleteAllProfile() {
        appRepository.deleteProfile()
    }

}